torch/_refs/linalg/__init__.py

```diff
@@ -113,6 +113,9 @@ def vector_norm(
     if isinstance(dim, Dim):
         dim = [dim]  # type: ignore[assignment]
 
+    if isinstance(ord, int) or isinstance(ord, torch.SymInt):
+        ord = float(ord)
```

Comment:
meh, this will force a specialization when you have a SymInt argument. This is not the worst thing to do, but it is not really necessary

Reply:
I am curious if it is SymInt, how do we suppose to deal with the predicates later on line 143-154 (`if ord == 0.0`, `if ord % 2.0 == 0.0`, etc )?

Reply:
Also, just wondering what will be the desired way to handling SymInt?

Reply:
Those predicates will add guards, which will introduce a recompilation if necessary.

Reply:
@ezyang I changed the predicate so that it handles `ord: Union[int, torch.SymInt]` and `ord: float` separately. This will fix the test issue, but I am not sure if we still want to fix the compiler so that it can handle `Union[int, torch.SymInt, float]` in one `ord % 2.0 == 0.0`.